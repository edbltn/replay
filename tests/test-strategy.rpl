strat {Rock, Paper, Scissors};

Strategy s = Strategy[3|4|{  state0: Rock, (Rock,Scissors) -> state1
                                      | (Rock,Rock) -> state2;
                             state1: Scissors, (_,_) -> state2;
                             state2: Paper, (Paper,_) -> state1
                                      | (Paper,Scissors) -> state0;}];

println(s.players);
println(s.moves);
println(s.size);
println(s);
