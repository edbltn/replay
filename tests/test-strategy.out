2
3
4
State 0: play 0
( 0 0 ) -> state 0
( 0 1 ) -> state 2
( 0 2 ) -> state 1
( 1 0 ) -> state 0
( 1 1 ) -> state 0
( 1 2 ) -> state 0
( 2 0 ) -> state 0
( 2 1 ) -> state 0
( 2 2 ) -> state 0
State 1: play 2
( 0 0 ) -> state 2
( 0 1 ) -> state 2
( 0 2 ) -> state 2
( 1 0 ) -> state 2
( 1 1 ) -> state 2
( 1 2 ) -> state 2
( 2 0 ) -> state 2
( 2 1 ) -> state 2
( 2 2 ) -> state 2
State 2: play 1
( 0 0 ) -> state 2
( 0 1 ) -> state 2
( 0 2 ) -> state 2
( 1 0 ) -> state 1
( 1 1 ) -> state 1
( 1 2 ) -> state 0
( 2 0 ) -> state 2
( 2 1 ) -> state 2
( 2 2 ) -> state 2
State 3: play 0
( 0 0 ) -> state 3
( 0 1 ) -> state 3
( 0 2 ) -> state 3
( 1 0 ) -> state 3
( 1 1 ) -> state 3
( 1 2 ) -> state 3
( 2 0 ) -> state 3
( 2 1 ) -> state 3
( 2 2 ) -> state 3

